client/containers/Pub/PubSyncManager.js

```diff
@@ -1,15 +1,17 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import queryString from 'query-string';
 
 import { loginDataProps } from 'types/base';
 import { apiFetch, getRandomColor } from 'utils';
-import queryString from 'query-string';
 import { initFirebase } from 'utils/firebaseClient';
+import { getPubPageTitle } from 'shared/utils/pubPageTitle';
 
 const propTypes = {
 	pubData: PropTypes.object.isRequired,
 	children: PropTypes.func.isRequired,
 	locationData: PropTypes.object.isRequired,
+	communityData: PropTypes.object.isRequired,
```

Comment:
Should this be pulling from PageWrapper context rather than being passed as a prop?

Reply:
The Context API is not ergonomic enough to use in a callback of a class-based React component. We'd either have to use a `<Consumer>` component and bend over backwards to capture the context passed into its function-child so that it can be referenced outside of `render()`, or rewrite the component in Hooks.

We really do need a `usePageContext()` hook though. It's like four lines of code and the next time I come up against this problem I'm going to implement it.

Reply:
Ah! Right - from the git diff view, I didn't see/remember this was still using React.Component.